src/main/java/com/hazelcast/kubernetes/KubernetesConfig.java

```diff
@@ -237,6 +252,18 @@ private void validateConfig() {
             throw new InvalidConfigurationException(
                     String.format("Property '%s' cannot be a negative number", SERVICE_PORT.key()));
         }
+        if (getMode().equals(DiscoveryMode.KUBERNETES_API)) {
+            String errorTemplate = "Property '%s' cannot be null when using kubernetes API mode";
```

Comment:
The error here does not necessarily mean that the property cannot be null. In most cases this property as `null` is perfectly fine, since it takes the value from the default file location.

How about rephrasing the error to something like:  "The value of CA Certificate cannot be found. Please make sure that the file '/var/run/secrets/kubernetes.io/serviceaccount/ca.crt' exists or define 'ca-certificate' in the configuration"